chain/jsonrpc-primitives/src/types/transactions.rs

```diff
@@ -0,0 +1,152 @@
+    InvalidTransaction { context: near_primitives::errors::InvalidTxError },
+    #[error("Node doesn't track this shard. Cannot determine whether the transaction is valid")]
+    DoesNotTrackShard,
+    #[error("Transaction with hash {transaction_hash} was routed")]
+    RequestRouted { transaction_hash: String },
+    #[error("Transaction {requested_transaction_hash} doesn't exist")]
+    UnknownTransaction { requested_transaction_hash: near_primitives::hash::CryptoHash },
+    #[error("The node reached its limits. Try again later. More details: {debug_info}")]
+    InternalError { debug_info: String },
+    #[error("Timeout")]
+    TimeoutError,
+    // NOTE: Currently, the underlying errors are too broad, and while we tried to handle
+    // expected cases, we cannot statically guarantee that no other errors will be returned
+    // in the future.
+    // TODO #3851: Remove this variant once we can exhaustively match all the underlying errors
+    #[error("It is a bug if you receive this error type, please, report this incident: https://github.com/near/nearcore/issues/new/choose. Details: {error_message}")]
+    Unreachable { error_message: String },
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RpcTransactionResponse {
+    #[serde(flatten)]
+    pub final_execution_outcome: near_primitives::views::FinalExecutionOutcomeViewEnum,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RpcBroadcastTxSyncResponse {
+    pub transaction_hash: String,
+    pub is_routed: bool,
+}
+
+impl RpcTransactionRequest {
+    pub fn parse(value: Option<Value>) -> Result<Self, crate::errors::RpcParseError> {
+        // TODO: move this code to crate::utils::parse_signed_transaction
+        let (encoded,) = crate::utils::parse_params::<(String,)>(value.clone())?;
+        let bytes = crate::utils::from_base64_or_parse_err(encoded)?;
+        let signed_transaction = near_primitives::transaction::SignedTransaction::try_from_slice(
+            &bytes,
+        )
+        .map_err(|err| {
+            crate::errors::RpcParseError(format!("Failed to decode transaction: {}", err))
+        })?;
+        Ok(Self { signed_transaction })
+    }
+}
+
+impl RpcTransactionStatusCommonRequest {
+    pub fn parse(value: Option<Value>) -> Result<Self, crate::errors::RpcParseError> {
+        if let Ok((hash, account_id)) =
+            crate::utils::parse_params::<(near_primitives::hash::CryptoHash, String)>(value.clone())
+        {
+            if !near_runtime_utils::is_valid_account_id(&account_id) {
+                return Err(crate::errors::RpcParseError(format!(
+                    "Invalid account id: {}",
+                    account_id
+                )));
+            }
+            let transaction_info = TransactionInfo::TransactionId { hash, account_id };
+            Ok(Self { transaction_info })
+        } else {
+            // TODO: move this code to crate::utils::parse_signed_transaction
```

Comment:
When TODO? I think it makes sense to inline `from_base64_or_parse_err` and extract this code into `parse_signed_transaction` as you suggest :+1: